### PR TITLE
feat: Manual scaling for trace and metric gateways

### PR DESCRIPTION
**Pull request by hisarbalik**
## Description

Changes proposed in this pull request:

- Add static scaling API for metric and trace gateway to Telemetry CR
- Add e2e test for static scaling API
- Add busola configuration for new API

## Related Issues and Documents

Closes:

Related issues:
https://github.com/kyma-project/telemetry-manager/issues/430

## Traceability

- [x] The PR is linked to a GitHub Issue.
- [ ] New features have a milestone label set.
- [x] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub Issue has a respective `area` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.

## Testability

The feature is unit-tested:

- [x] Yes.
- [ ] No, because unit tests are not needed.
- [ ] No, because of ...

The feature is e2e-tested:

- [x] Yes.
- [ ] No, because e2e-tests are not needed.
- [ ] No, because of ...

Tests conducted for the PR:

## Codebase

- [x] My code follows the [Effective Go](https://go.dev/doc/effective_go) style guidelines.
- [x] The code was planned and designed following the defined architecture and the separation of concerns.
- [ ] The code has sufficient comments, particularly for all hard-to-understand areas.
- [x] This PR adds value and shows no feature creep.
- [ ] I have augmented the test suite that proves my fix is effective or that my feature works.
- [ ] Adjusted the documentation if the change is user-facing.

### CODE REVIEW

File: docs/user/resources/01-telemetry.md [hunk 1]
```diff
@@ -57,6 +57,23 @@ For details, see the [Telemetry specification file](https://github.com/kyma-proj
...
 
+**Spec:**
+
+| Parameter | Type | Description |
+| ---- | ----------- | ---- |
+| **metric**  | object | MetricSpec is the specification of the desired behavior of the metric gateway |
+| **metric.&#x200b;gateway**  | object |  |
+| **metric.&#x200b;gateway.&#x200b;scaling**  | object | Scaling defines desired type scaling and number of desired pods. |
+| **metric.&#x200b;gateway.&#x200b;scaling.&#x200b;static**  | object | Static scaling config params. Present only if Type = StaticScalingStrategyType. |
+| **metric.&#x200b;gateway.&#x200b;scaling.&#x200b;static.&#x200b;replicas**  | integer |  |
```
Comment (NHingerl) [inline docs/user/resources/01-telemetry.md:68]: Why is this missing a description?

File: apis/operator/v1alpha1/telemetry_types.go [hunk 2]
```diff
@@ -38,8 +38,51 @@ const (
...
+// TraceSpec is the specification of the desired behavior of the trace gateway
+type TraceSpec struct {
+	Gateway TraceGatewaySpec `json:"gateway,omitempty"`
+}
+
+type TraceGatewaySpec struct {
+	Scaling Scaling `json:"scaling,omitempty"`
+}
+
+// Scaling defines desired type scaling and number of desired pods.
```
Comment (NHingerl) [inline apis/operator/v1alpha1/telemetry_types.go:63]: Do we even need the word "desired"?

```suggestion
// Scaling defines the scaling type and number of pods.
```

File: apis/operator/v1alpha1/telemetry_types.go [hunk 3]
```diff
@@ -38,8 +38,51 @@ const (
 
 // TelemetrySpec defines the desired state of Telemetry
 type TelemetrySpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	Trace  TraceSpec  `json:"trace,omitempty"`
+	Metric MetricSpec `json:"metric,omitempty"`
+}
+
+// MetricSpec is the specification of the desired behavior of the metric gateway
```
Comment (NHingerl) [inline apis/operator/v1alpha1/telemetry_types.go:45]:
```suggestion
// MetricSpec defines the behavior of the metric gateway
```

File: apis/operator/v1alpha1/telemetry_types.go [hunk 4]
```diff
@@ -38,8 +38,51 @@ const (
...
+type MetricSpec struct {
+	Gateway MetricGatewaySpec `json:"gateway,omitempty"`
+}
+
+type MetricGatewaySpec struct {
+	Scaling Scaling `json:"scaling,omitempty"`
+}
+
+// TraceSpec is the specification of the desired behavior of the trace gateway
```
Comment (NHingerl) [inline apis/operator/v1alpha1/telemetry_types.go:54]:
```suggestion
// TraceSpec defines the behavior of the trace gateway
```

File: internal/resources/otelcollector/gateway/resources.go [hunk 5]
```diff
@@ -42,6 +42,16 @@ type ServiceConfig struct {
 	OTLPServiceName string
 }
 
+type Scaling struct {
+	// Replicas specifies the desired number of gateway replicas.
```
Comment (NHingerl) [inline internal/resources/otelcollector/gateway/resources.go:46]:
```suggestion
	// Replicas specifies the number of gateway replicas.
```

File: internal/resources/otelcollector/gateway/resources.go [hunk 6]
```diff
@@ -42,6 +42,16 @@ type ServiceConfig struct {
...
+	Replicas int32
+
+	// ResourceRequirementsMultiplier is a coefficient affecting the CPU and memory resource limits for each replica.
+	// This value is multiplied with a base resource requirement to calculate the actual CPU and memory limits.
+	// A value of 1 will apply the base limits, values greater than 1 will proportionally increase those limits.
```
Comment (NHingerl) [inline internal/resources/otelcollector/gateway/resources.go:51]:
```suggestion
	// A value of 1 applies the base limits; values greater than 1 increase those limits proportionally.
```

File: docs/user/resources/01-telemetry.md (hunk 1 above)
Comment (hisarbalik) [reply docs/user/resources/01-telemetry.md]: description added

**Review by NHingerl — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Pull request merged**